src/llm/apis/openai_completions.cpp

```diff
@@ -191,7 +195,13 @@ absl::Status OpenAIChatCompletionsHandler::parseMessages() {
                             return absl::InvalidArgumentError("Unsupported content type");
                         }
                     }
+                    // Pulling out text from nested structure to the "content" field for text and erase whole "content" value for image data
+                    // since images are stored separately in request.images
                     member->value = contentText;
+                    // Add new field to the last message in history if content is text
+                    if (!member->value.IsNull() && member->value.IsString()) {
```

Comment:
should we error error if its different type?

Reply:
It's actually only additional safety layer. Type check of the content we end up with is checked earlier as we extract it.